src/Xrefcheck/Verify.hs

```diff
@@ -327,5 +360,49 @@ checkExternalResource VerifyConfig{..} link
         HttpExceptionRequest _ exc -> case exc of
           StatusCodeException resp _
             | isAllowedErrorCode (statusCode $ responseStatus resp) -> Right ()
-            | otherwise -> Left $ ExternalResourceUnavailable (responseStatus resp)
+            | otherwise -> Left $ ExternalHttpResourceUnavailable (responseStatus resp)
           other -> Left . ExternalResourceSomeError $ show other
+
+    checkFtp :: URI -> Bool -> ExceptT VerifyError IO ()
+    checkFtp uri secure = do
+      -- get authority which stores host and port
+      authority <- case uriAuthority uri of
+        Right a -> pure a
+        Left _ -> throwError $
+          ExternalResourceInvalidUrl (Just "FTP path must be absolute")
+      let host = T.unpack . unRText $ authHost authority
+      let port :: Int = fromIntegral . fromMaybe 21 $ authPort authority
+      -- build path from pieces
+      path <- case uriPath uri of
+        Nothing -> pure "/"
+        Just (_, pieces) -> pure . T.unpack . mconcat
+          . NonEmpty.toList
+          . NonEmpty.intersperse "/"
+          . NonEmpty.map unRText
+          $ pieces
```

Comment:
Quite annoying that the library does not provide an utility for this rendering :unamused: 

By the way, `T.unpack` is available in our prelude as `toString`, and `NonEmpty.toList` - probably just as `toList`. Since you already add a module with orphans, maybe it would be convenient to write your own `ToString (RText t)` instance and avoid calling `unRText` explicitly each time.

Reply:
good idea

Reply:
Looks pretty now :+1: